modules/renter/repair.go

```diff
@@ -193,7 +194,11 @@ func (r *Renter) managedRepairIteration(rs *repairState) {
 		}
 
 		// Ignore workers that have had an upload failure recently.
-		if time.Since(worker.recentUploadFailure) < uploadFailureCooldown {
+		penalty := uint64(worker.consecutiveUploadFailures)
+		if worker.consecutiveUploadFailures > maxConsecutivePenalty {
+			penalty = uint64(maxConsecutivePenalty)
+		}
+		if time.Since(worker.recentUploadFailure) < uploadFailureCooldown*(1<<penalty) {
```

Comment:
Easier to tweak but more annoying to implement. This overall felt pretty clean.

If we add the cooldown to the worker, then we need to change the recent failure mechanic from being a recent failure time to a 'cooldown expires time'.

I'll touch it up.

I'm also not really sure how to test this, I don't think we have a good framework yet for putting the renter with offline hosts.